### PR TITLE
feat(api) : project initialized for api

### PR DESCRIPTION
**Pull request by soumyadip007**
## Project Intialization

1. Project Initialized for API

## Screenshot(s)

<details>
<summary>View Screenshots</summary>

</details>

### Ready-for-merge Checklist

- [ ] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [ ] Does the change have appropriate unit tests?
- [ ] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [ ] Was this feature demo'd and the design review approved?

### CODE REVIEW

File: packages/api/Dockerfile [hunk 1]
```diff
@@ -36,6 +36,8 @@ RUN find . -name "node_modules" -exec rm -rf '{}' +
 RUN rm -rf '/app/package-lock.json'
 
 RUN sudo apt-get install nodejs
+RUN sudo npm install -g npm-check-updates
```
Comment (hybridx) [inline packages/api/Dockerfile:39]: Do we need check updates? Because the new update might break the code

File: packages/api/package.json [hunk 2]
```diff
@@ -1,81 +1,102 @@
...
+    "start:dev": "nest start --watch",
+    "start:debug": "nest start --debug --watch",
+    "start:prod": "node dist/main",
+    "lint": "eslint --fix --ext ts ./src",
+    "test": "jest",
+    "test:watch": "jest --watch",
+    "test:cov": "jest --coverage",
+    "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
+    "test:e2e": "jest --config ./test/jest-e2e.json"
   },
```
Comment (hybridx) [inline packages/api/package.json:20]: @soumyadip007 you can let the old contributors still be there

**Review by arkaprovob — CHANGES_REQUESTED** [review]
Move the exception folder and it's contents from `src/configuration/` to `/src/server/`

File: packages/api/src/configuration/exceptions/exception.abstract.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+export abstract class IFormatExceptionMessage {
```
Comment (hybridx) [inline packages/api/src/configuration/exceptions/exception.abstract.ts]: @soumyadip007 nest kindha follows angular pattern hence the exception should be ether in their respective files / controllers or insider services

File: packages/api/src/server/property/envoronment.dto.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { ApiProperty, PartialType } from "@nestjs/swagger";
+import { Application } from "src/server/application/application.entity";
+
```
Comment (akhilmhdh) [inline packages/api/src/server/property/envoronment.dto.ts:3]: File typo

**Comment by soumyadip007** [comment]
Hi @arkaprovob , I've made the changes please check.

File: packages/api/src/configuration/index.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export const DATA_BASE_CONFIGURATION = {
```
Comment (akhilmhdh) [inline packages/api/src/configuration/index.ts:1]: Use nest/env with joi validation

**Review by akhilmhdh — CHANGES_REQUESTED** [review]

File: packages/api/src/server/property/envoronment.dto.ts (hunk 4 above)
Comment (soumyadip007) [reply packages/api/src/server/property/envoronment.dto.ts]: Done

**Review by hybridx — APPROVED** [review]
minor change rest LGTM

**Comment by hybridx** [comment]
We need to also check this with the new manager, once this code is merged into spaship develop then we can make changes for manager as well

File: packages/api/Dockerfile (hunk 1 above)
Comment (soumyadip007) [reply packages/api/Dockerfile:39]: There was an issue after the deployment, that why we've added it. Let's keep it as of now, will remove it later if it's not needed after testing.

File: packages/api/src/configuration/exceptions/exception.abstract.ts (hunk 3 above)
Comment (soumyadip007) [reply packages/api/src/configuration/exceptions/exception.abstract.ts]: Yes I've already moved that.

**Review by arkaprovob — APPROVED** [review]

File: packages/api/src/configuration/index.ts (hunk 5 above)
Comment (soumyadip007) [reply packages/api/src/configuration/index.ts:1]: As of now we can ignore the nest/env. now that is only DB config. I'll add it shortly.

**Pull request merged**